src/gui/sdlmain.cpp

```diff
@@ -1741,32 +1741,29 @@ static void check_kmsdrm_setting()
 	exit(1);
 }
 
-// Inform the VGA side if it should draw double-scanned lines or not (ie:
-// single-scan) at rendering time. We only allow single-scanning for output
-// modes that treat pixels as sharp little rectangles because they produce
-// identical visual results in both scenarios.
+// Double-scanning criteria:
 //
-// If the visual results are the same, can we simplify this by double-scanning
-// all the time? In the future, yes! -- but for now, single scanning consumes
-// 1/4th the bandwidth, which retains meaningful performance on
-// bandwidth-limited single board computers (SBCs) like the Raspberry Pi 4.
+//  - If OpenGL is using any shader (regardless of mode)
+//  - If the output mode is bilinear scaled
```

Comment:
So this has nothing to do with `openglnb` right? Practically all OpenGL shaders that Tyrell converted use a texture sampling helper function that guarantees identical output in both `opengl` and `openglnb` modes (in`openglnb`, the bilinear scaling is emulated via extra shader code). The shader has that info in which OpenGL mode it's running via preprocessor defines.

Reply:
Right; all output modes (opengl, nb, and pp) are double-scanned when a shader's being used.